vitrina/datasets/views.py

```diff
@@ -85,20 +86,26 @@ def get_context_data(self, **kwargs):
         return context
 
 
-class DatasetDetailView(DetailView):
+class DatasetDetailView(LanguageChoiceMixin, DetailView):
     model = Dataset
     template_name = 'vitrina/datasets/detail.html'
     context_object_name = 'dataset'
 
     def get_context_data(self, **kwargs):
         context_data = super().get_context_data(**kwargs)
         dataset = context_data.get('dataset')
+        language = self.request.GET.get('language', '')
+        if language:
+            dataset.set_current_language(language)
+        else:
+            dataset.set_current_language(settings.LANGUAGE_CODE)
```

Comment:
For some reason, when I switch to English language and I click on a dataset from dataset list it always switches to Lithuanian language, which is set as default. So I think line is responsible for that.

I don't understand why you need that?